### PR TITLE
Add more unit tests for empty urls

### PR DESCRIPTION
**Pull request by SergioEstevao**
Add more unit tests for empty urls.

Needs Review: @bummytime

### CODE REVIEW

File: Example/Tests/WordPressXMLRPCApiTests.m [hunk 1]
```diff
@@ -18,12 +18,40 @@ - (void)tearDown {
...
-        [expectation fulfill];
+        NSLog(@"%@", [error localizedDescription]);
+        [expectationEmpty fulfill];
+        errorToCheck = error;
     }];
     [self waitForExpectationsWithTimeout:2 handler:nil];
+    XCTAssertTrue(errorToCheck.domain == WordPressXMLRPCApiErrorDomain, @"Expected to get an WordPressXMLRPCApiErrorDomain error");
+    XCTAssertTrue(errorToCheck.code == WordPressXMLRPCApiEmptyURL, @"Expected to get an WordPressXMLRPCApiEmptyURL error");
+
+    XCTestExpectation *expectationNil = [self expectationWithDescription:@"Call should fail with error when invoking with nil string"];
```
Comment (bummytime) [inline Example/Tests/WordPressXMLRPCApiTests.m]: Maybe a style thing, but would it be possible to split this test into 3 separate ones to keep them smaller and more focused?

**Comment by SergioEstevao** [comment]
@bummytime I always struggle with this, one test method for each method being tested or for each scenario? Do you have any rational

**Comment by bummytime** [comment]
Yeah I am right there with you - I never seem to know the balance. The thing I always ask myself is "Does each unit test have a single reason to fail?" IIRC I read this years ago in one of the Pragmatic Programmer books. So related to the test above it could be argued that the test has 3 different reasons to fail.

I am not dogmatic about it though :-) Ultimately you are testing the right things, so if you don't feel a strong need to change it.... :shipit:
